### PR TITLE
Fix name suggestions (freshen default name)

### LINKED ISSUE #2919: "Extract variable" refactoring renames usages of an accidentally name-conflicting variable
## Environment

* **Intellij-Rust plugin version:**
Latest git (b1ce136d7332a2b031688e00443625ce076ed904)
* **Rust toolchain version:**
rustc 1.31.0-nightly (2bd5993ca 2018-10-02)
* **IDE name and version:**
Idea Community 2018.2.4 (IC-182.4505.22)
* **Operating system:**
Linux x86-64

## Problem description
When selecting an expression and executing "Extract variable" refactoring, a new variable with the generic name, e.g. "x" is introduced and its name is highlighted in such way, that editing it also changes its name at the point of usage, where there has been the extracted expression. 

But if a variable with such name have already existed, any typing attempt changes all its usages too, not just at the place of extracted expression.  

The refactoring apparently tries to avoid that by choosing "i" or "x", whichever is currently unoccupied, but if both are present, it just gives up and uses "x", with ensuing incorrect behavior.

## Steps to reproduce

1. Take this simple program.
1. Select "3".
1. Execute the refactoring.
1. Type to rename the new variable to "blah".

```rust
fn main() {
    let i = 1; 
    let x = 2;

    let z = 3 + 4;
    let w = x + 5;
}
```
### Expected result
```rust
fn main() {
    let i = 1; 
    let x = 2;
    let blah = 3;

    let z = blah + 4;
    let w = x + 5;
}
```
### Actual result
```rust
fn main() {
    let i = 1; 
    let x = 2;
    let blah = 3;

    let z = blah + 4;
    let w = blah + 5;
}
```

Note how `let w = x + 5` became `let w = blah + 5`, even though it had nothing to do with any of it.

### PR DESCRIPTION
**Pull request by ortem**
Fixes #2919

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt [hunk 1]
```diff
@@ -91,9 +104,9 @@ private fun nameForCall(expr: RsCallExpr): List<String> {
     return listOf(pathElement.text)
 }
 
-private fun findNamesInLocalScope(expr: PsiElement): List<String> {
+private fun findNamesInLocalScope(expr: PsiElement): HashSet<String> {
     val blockScope = expr.ancestorOrSelf<RsBlock>()
     val letDecls = blockScope?.descendantsOfType<RsLetDecl>().orEmpty()
 
-    return letDecls.mapNotNull { it.pat?.text }
+    return letDecls.asSequence().mapNotNull { it.pat?.text }.toHashSet()
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt]: Hahahaha

Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt]: let (i, x) = (1, 2);

File: src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt [hunk 2]
```diff
@@ -91,9 +104,9 @@ private fun nameForCall(expr: RsCallExpr): List<String> {
     return listOf(pathElement.text)
 }
 
-private fun findNamesInLocalScope(expr: PsiElement): List<String> {
+private fun findNamesInLocalScope(expr: PsiElement): HashSet<String> {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt]: `Set` can be used?

File: src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt (hunk 1 above)
Comment (Undin) [reply src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt]: I still think that in such cases we should use `mapNotNullTo`

**Comment by ortem** [comment]
bors retry

**Pull request merged**